app/pages/error-500.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import Link from 'next/link';
+import Layout from 'components/Layout';
+import { withRelay, RelayProps } from 'relay-nextjs';
+import { usePreloadedQuery } from 'react-relay/hooks';
+import { graphql } from 'react-relay';
+import defaultRelayOptions from 'lib/relay/withRelayOptions';
+import { error500Query } from '__generated__/error500Query.graphql';
+
+const getError500Query = graphql`
+  query error500Query {
+    session {
+      sub
+    }
+  }
+`;
+const Error500Page = ({
```

Comment:
You may want to consider making this into a static `pages/500.tsx` page that doesn't use a relay query (and doesn't pass the `session` to `Layout`).
It would be rendered for every 500 error and replace the generic nextjs error page.

Reply:
I considered that, I will have to make a new static layout that doesn't have a login link if there is no session as it will be confusing if there is an error for a logged in user.